fl4health/feature_alignment/handle_types.py

```
@@ -0,0 +1,592 @@
+    return to_dtype(series, NUMERIC), {FEATURE_TYPE_ATTR: NUMERIC}
+
+
+def _convertible_to_categorical(
+    series: pd.Series,
+    category_min: int | None = None,
+    category_max: int | None = None,
+    unique: np.ndarray | None = None,
+    raise_error_over_max: bool = False,
+    raise_error_under_min: bool = False,
+) -> bool:
+    """
+    Check whether a feature can be converted to some categorical type.
+
+    Args:
+        series (pd.Series): Feature data.
+        category_min (int | None, optional):  The minimum number of categories allowed. Defaults to None.
+        category_max (int | None, optional): The maximum number of categories allowed. Defaults to None.
+        unique (np.ndarray | None, optional): Unique values which can be optionally specified. Defaults to None.
+        raise_error_over_max (bool, optional):  Whether to raise an error if there are more categories than max.
+            Defaults to False.
+        raise_error_under_min (bool, optional): Whether to raise an error if there are less categories than min.
+            Defaults to False.
+
+    Raises:
+        ValueError: Raise an error if there are more categories than max and ``raise_error_over_max`` is True
+        ValueError: Raise an error if there are less categories than min and ``raise_error_under_min`` is True
+
+    Returns:
+        bool: Whether the feature can be converted.
+    """
+    # If numeric, only allow conversion if an integer type
+    if is_numeric_dtype(series) and not is_integer_dtype(series):
+        return False
+
+    unique = get_unique(series, unique=unique)
+    nonnull_unique = unique[~pd.isnull(unique)]
+    nunique = len(nonnull_unique)
+
+    min_cond = True if category_min is None else nunique >= category_min
+
+    max_cond = True if category_max is None else nunique <= category_max
+
+    # Convertible
+    if min_cond and max_cond:
+        return True
+
+    # Not convertible
+    if max_cond and raise_error_over_max:
```

Isn't this logic flipped? `max_cond` checks if the number of unique values is less than the category maximum. If we want to raise an error if the number of values is larger than the max, shouldn't this be `if not max_cond and raise_error _over_max`? Same with `min_cond`.

Also, changing the name of `max_cond` and `min_cond` to something more descriptive would make this easier to spot. Something like `within_max` or `satisfies_max` maybe?